### PR TITLE
fix: terminate plan when deleting public plan proposal

### PR DESCRIPTION
**Pull request by kogisin**
## Description

closes: #161 

## Tasks

- [x] Add `TerminationPlan` logic in `DeletePublicPlanProposal` 
     * Handle an edge case where `FarmingPoolAddress` and `TerminationAddress` are equal
- [x] Add test code to cover the case
- [x] Update related docs

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/farming/keeper/proposal_handler.go [hunk 1]
```diff
@@ -225,6 +225,12 @@ func (k Keeper) DeletePublicPlanProposal(ctx sdk.Context, proposals []*types.Del
 			return sdkerrors.Wrapf(sdkerrors.ErrNotFound, "plan %d is not found", p.GetPlanId())
 		}
 
+		if plan.GetFarmingPoolAddress().String() != plan.GetTerminationAddress().String() {
```
Comment (dongsam) [inline x/farming/keeper/proposal_handler.go:228]: even if a plan have same `TerminationAddress` with `FarmingPoolAddress` the plan should be terminated with emitting events and setting `plan.terminated` to `true`

so, I think this if statements should be on `TerminatePlan` below line https://github.com/tendermint/farming/blob/783b9420577443d9aa020cb4f781f05382bed918/x/farming/keeper/plan.go#L244

File: x/farming/keeper/proposal_handler_test.go [hunk 2]
```diff
@@ -406,80 +406,120 @@ func (suite *KeeperTestSuite) TestValidateDeletePublicPlanProposal() {
...
+	suite.Require().Equal(initialBalances, poolInitialBalances)
+
+	// delete the plan
+	err = keeper.HandlePublicPlanProposal(
+		suite.ctx,
+		suite.keeper,
+		types.NewPublicPlanProposal("testTitle", "testDescription", nil, nil, []*types.DeleteRequestProposal{
+			types.NewDeleteRequestProposal(plan.GetId()),
+		}),
+	)
```
Comment (dongsam) [inline x/farming/keeper/proposal_handler_test.go:520]: Could you add a test case to asserting even if a plan has the same `TerminationAddress` with `FarmingPoolAddress` the plan set terminated

Comment (kogisin) [reply x/farming/keeper/proposal_handler_test.go:520]: I can add the test case to cover when `TerminationAddress` with `FarmingPoolAddress` are the same. However, as plan gets removed from the store, I am not sure if there is a way to check whether or not the Terminated is set to true.

Comment (dongsam) [reply x/farming/keeper/proposal_handler_test.go]: Yes, we can't check the plan state after deletion, but we can check the successfully removed with balance differences using `farming.EndBlocker(suite.ctx, suite.keeper)` on test code after handle `PublicPlanProposal` and also events like https://github.com/cosmos/cosmos-sdk/blob/d60cb3a49f73c8f51da80076d22d3553c5a8b20b/x/bank/keeper/keeper_test.go#L1087-L1113

**Review by dongsam — APPROVED** [review]
